tests/integration/questionnaire/test_questionnaire_last_viewed_guidance_hub.py

```diff
@@ -0,0 +1,226 @@
+        response_id = self.random_respondent_id()
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # When I sign out and resume on a primary person list collector add person page
+        self.post()
+        self.post_address_confirmation_answer()
+        self.post_you_live_here_answer()
+        self.post(action="sign_out")
+
+        # Then the last viewed guidance is shown on the parent page
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+        self.assertInUrl(
+            "/questionnaire/primary-person-list-collector/?last_viewed_question_guidance=True"
+        )
+        self.last_viewed_question_guidance_shown_assertion(
+            "/questionnaire/household-interstitial/"
+        )
+
+    def test_last_viewed_question_guidance_shown_on_section_in_progress_resume_list_collector(
+        self
+    ):
+
+        # Given
+        response_id = self.random_respondent_id()
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # When I sign out and resume on a list collector
+        self.post()
+        self.post_address_confirmation_answer()
+        self.post_you_live_here_answer()
+        self.post_primary_person_answer()
+        self.post(action="sign_out")
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # Then the last viewed guidance is shown
+        self.assertInUrl(
+            "/questionnaire/list-collector/?last_viewed_question_guidance=True"
+        )
+        self.last_viewed_question_guidance_shown_assertion(
+            "/questionnaire/household-interstitial/"
+        )
+
+    def test_last_viewed_question_guidance_shown_on_section_in_progress_resume_list_collector_add_person(
+        self
+    ):
+
+        # Given
+        response_id = self.random_respondent_id()
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # When I sign out and resume on a list collector add person
+        self.post()
+        self.post_address_confirmation_answer()
+        self.post_you_live_here_answer()
+        self.post_primary_person_answer()
+        self.post({"anyone-else": "Yes"})
+        self.post(action="sign_out")
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
+
+        # Then the last viewed guidance is shown on the parent page
+        self.assertInUrl(
+            "/questionnaire/list-collector/?last_viewed_question_guidance=True"
+        )
+        self.last_viewed_question_guidance_shown_assertion(
+            "/questionnaire/household-interstitial/"
+        )
+
+    def test_last_viewed_question_guidance_shown_on_section_in_progress_resume_relationships(
+        self
+    ):
+
+        # Given
+        response_id = self.random_respondent_id()
+        self.launchSurvey("test_last_viewed_question_guidance", reponse_id=response_id)
```

Comment:
It would be worth thinking about a common method that does the response id generation and returns the generated id.